### ISSUE TITLE
[DE223]: Create Volume from Image Fails

### ISSUE THREAD

**Issue by byronmccollum** [rcbops/chef-cookbooks#489]
#### This is an issue in the `v4.0.0` branch of the cookbooks, which is the default for RPCS. Please do not close saying it was fixed in a different brach.

---

When doing something like the following, the volume create fails:

```
cinder create 2 --image-id <image-id> --display-name MyVolume
```

The Cinder-Volume nodes need `glance_host` defined in `cinder.conf` to know where to pull the image from.

Edit: This was fixed in [#49](https://github.com/rcbops-cookbooks/cinder/pull/49) and [#56](https://github.com/rcbops-cookbooks/cinder/pull/56) as noted by @DavidWittman below, but needs backporting.

**Comment by byronmccollum** [comment]
Error from `cinder-volume.log`:

```
ERROR [cinder.image.glance] Error contacting glance server '10.0.1.105:9292' for 'get', done trying.
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/cinder/image/glance.py", line 146, in call
    return getattr(client.images, method)(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/glanceclient/v1/images.py", line 104, in get
    % urllib.quote(image_id))
  File "/usr/lib/python2.7/dist-packages/glanceclient/common/http.py", line 260, in raw_request
    return self._http_request(url, method, **kwargs)
  File "/usr/lib/python2.7/dist-packages/glanceclient/common/http.py", line 207, in _http_request
    raise exc.CommunicationError(message=message)
CommunicationError: Error communicating with http://10.0.1.105:9292 [Errno 111] ECONNREFUSED
2013-07-23 10:42:06    ERROR [cinder.volume.manager] volume volume-6e07095c-7db5-43be-a547-65becaae941a: create failed
2013-07-23 10:42:06    ERROR [cinder.openstack.common.rpc.amqp] Exception during message handling
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/cinder/openstack/common/rpc/amqp.py", line 430, in _process_data
    rval = self.proxy.dispatch(ctxt, version, method, **args)
  File "/usr/lib/python2.7/dist-packages/cinder/openstack/common/rpc/dispatcher.py", line 133, in dispatch
    return getattr(proxyobj, method)(ctxt, **kwargs)
  File "/usr/lib/python2.7/dist-packages/cinder/volume/manager.py", line 288, in create_volume
    LOG.error(_("volume %s: create failed"), volume_ref['name'])
  File "/usr/lib/python2.7/contextlib.py", line 24, in __exit__
    self.gen.next()
  File "/usr/lib/python2.7/dist-packages/cinder/volume/manager.py", line 238, in create_volume
    image_location = image_service.get_location(context, image_id)
  File "/usr/lib/python2.7/dist-packages/cinder/image/glance.py", line 220, in get_location
    _reraise_translated_image_exception(image_id)
  File "/usr/lib/python2.7/dist-packages/cinder/image/glance.py", line 218, in get_location
    image_meta = client.call(context, 'get', image_id)
  File "/usr/lib/python2.7/dist-packages/cinder/image/glance.py", line 159, in call
    reason=str(e))
GlanceConnectionFailed: Connection to glance failed: Error communicating with http://10.0.1.105:9292 [Errno 111] ECONNREFUSED
```

**Comment by DavidWittman** [comment]
@byronmccollum Are you using older cookbooks? This should have been fixed in rcbops-cookbooks/cinder#49 and rcbops-cookbooks/cinder#56.

**Comment by mattt416** [comment]
Hi @byronmccollum,

Thanks for creating this issue.  We are currently reviewing the best way of getting such fixes from our grizzly branch into the v4.0.0 branch and will update this ticket with more information once we have something nailed down and implemented.

Thanks,
Matt

**Comment by jcannava** [comment]
This was fixed in This was fixed in #49 and #56 as mentioned by @byronmccollum and @DavidWittman. These fixes are now included in the new default tag for RPC v4.1.2 so closing.
